packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -952,6 +952,36 @@ describe('TransactionController', () => {
       );
     });
 
+    it('adds initial history', async () => {
+      const controller = newController();
+
+      await controller.addTransaction({
+        from: ACCOUNT_MOCK,
+        to: ACCOUNT_MOCK,
+      });
+
+      expect(controller.state.transactions[0]?.history).toBeDefined();
+      // Atleast 1 history item (initial snapshot)
+      expect(controller.state.transactions[0]?.history?.length).toBeGreaterThan(
+        0,
+      );
+    });
+
+    it('doesnt populate history if history explicitly disabled', async () => {
```

Comment:
As we're not testing the utils, should we also have a specific test to ensure at least one of the public methods updates the history as we expect?

We could even make a test template function, and reuse it on all the public methods that update the history to be as thorough as possible?

Reply:
Good call, I've tried but couldn't find an easy way to make it possible with template function because different public methods require it's own logic and requests (to check all possible history updates), then the template function gets messy. 

So I find it easier to include tests which at least checking initial history generation. So I've added for `confirmExternalTransaction` & `addTransaction` which I believe most important pieces since they are entry points for transactions.

Pushed here: https://github.com/MetaMask/core/pull/1657/commits/7a071b74b1a9eb7d37de6115cd141189bbc8911b